### PR TITLE
Ft two factor authentication for seller 187355104

### PR DESCRIPTION
**Pull request by venuste-20002**
#### What does this PR do?
   
 The Two Factor Authentication for Sellers

#### Description of Task to be completed?

      This Task  Have the following Endpoint
  1. Endpoint: POST /toggle-2fa
   
          Description: This endpoint will allow a seller to enable or disable two-factor authentication.
            When a request is made to this endpoint, 
            the system will toggle the current state of 2FA for the specified user.

        Implementation Steps:
             .Fetch the user from the database using the provided userId.
             .If the user is not found, return a 404 Not Found response.
             .Toggle the use2FA field for the user.
             .Save the updated user information in the database.
             .Return a 200 OK response with a message indicating the new status of 2FA.

  2. Endpoint: POST /2fa-verify
          
      Description: This endpoint will verify the provided 2FA token for a seller. 
       If the token is valid, a JWT token is generated and returned.

      Implementation Steps:
         . Fetch the user from the database using the provided userId.
        . If the user is not found, return a 404 Not Found response.
        .Verify the provided 2FA token using the secret stored in the user's record.
        .If the token is valid, generate a JWT token and return it in the response.
        .If the token is invalid, return a 401 Unauthorized response with an appropriate message.

#### How should this be manually tested?

     after clone and install all the independences you used
     Swagger Documentation with the above end point to  test

#### What are the relevant pivotal tracker/Trello stories?

https://www.pivotaltracker.com/story/show/187355104

#### Screenshots (if appropriate)

Screenshot 2024-05-22 174324

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [![HoundCI](https://img.shields.io/badge/reviewed%20by-Hound-%23a873d1.svg)](https://houndci.com)
-[![Coverage Status](https://coveralls.io/repos/github/atlp-rwanda/e-commerce-ones-and-zeroes-bn/badge.svg?branch=develop)](https://coveralls.io/github/atlp-rwanda/e-commerce-ones-and-zeroes-bn?branch=develop)
+[![Coverage Status](https://coveralls.io/repos/github/atlp-rwanda/e-commerce-ones-and-zeroes-bn/badge.svg?branch=ft-setRoles-187355108)](https://coveralls.io/github/atlp-rwanda/e-commerce-ones-and-zeroes-bn?branch=ft-setRoles-187355108)
 
-[![CircleCI](https://circleci.com/gh/atlp-rwanda/e-commerce-ones-and-zeroes-bn/tree/develop.svg?style=svg)](https://app.circleci.com/pipelines/github/atlp-rwanda/e-commerce-ones-and-zeroes-bn/?branch=develop)
+[![CircleCI](https://circleci.com/gh/atlp-rwanda/e-commerce-ones-and-zeroes-bn/tree/develop.svg?style=svg)](https://app.circleci.com/pipelines/github/atlp-rwanda/e-commerce-ones-and-zeroes-bn/?branch=ft-setRoles-187355108)
```
Comment (UmuhireJessie) [inline README.md]: Please do not change these, we need to see the coverage and CI statuses on `develop` branch

File: src/controllers/sellerController.ts
Comment (UmuhireJessie) [inline src/controllers/sellerController.ts:1]: Can you simplify the process based on what we agreed? Or, if you choose to continue with this, explain it in the Readme file and the description of the PR

File: src/database/seeders/20240415080213-demo-user.ts [hunk 2]
```diff
@@ -1,56 +1,42 @@
...
       [
         {
           userId: uuid(),
           firstName: 'christian',
           lastName: 'Ishimwe',
-          email: 'iigore2061@gmail.com',
-          password: hashedPassword1,
-          isVerified: true,
+          email: 'christianinja3@gmail.com',
+          password: process.env.password,
```
Comment (UmuhireJessie) [inline src/database/seeders/20240415080213-demo-user.ts]: A password should be hashed before being saved to the database, regardless of it being an environment variable

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
So far, so good @venuste-20002, please:
1. work on the comments
2. squash the commits
3. resolve the conflicts

**Review by UmuhireJessie — APPROVED** [review]

File: src/documents/1716834651217-WIN_20231125_19_36_42_Pro.jpg
Comment (UmuhireJessie) [inline src/documents/1716834651217-WIN_20231125_19_36_42_Pro.jpg:1]: What are these images for?

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
@venuste-20002 Check my comments

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Good @venuste-20002, please resolve conflicts

**Review by lilika67 — APPROVED** [review]

**Review by shema-surge — APPROVED** [review]

**Review by UmuhireJessie — APPROVED** [review]

**Pull request merged**